packages/@aws-cdk/core/lib/environment-component.ts

```diff
@@ -0,0 +1,130 @@
+import { Token } from './token';
+
+/**
+ * Interface representing the environment a given resource lives in.
+ * Used as the return value for the IResource.environment property.
+ */
+export interface ResourceEnvironment {
+  /** The AWS account ID that this resource belongs to. */
+  readonly account: EnvComponent;
```

Comment:
What's special about `EnvComponent`. It's just a string that can be a token. Can we generalize this into `StringToken` or something?

Reply:
Got rid of this class completely in favor of comparing strings in `Token.compareStrings()`.